classes/manager.php

```diff
@@ -345,26 +345,25 @@ public function is_considered_slow(profile $profile): bool {
 
     /**
      * Increments a counter using the approximate counting algorithm.
-     * Can handle up to PHP_INT_SIZE_BITS counts (2^PHP_INT_SIZE_BITS events).
      *
      * See https://en.wikipedia.org/wiki/Approximate_counting_algorithm
      *
      * @param int $current The current count.
+     * @param int $increment The raw increment in count.
      * @return int The new count.
      */
-    public static function approximate_increment(int $current): int {
-        // If the number of events is ever expected to be more than 2 billion, a refactor may be needed.
-
-        $bits = random_int(PHP_INT_MIN, PHP_INT_MAX);
-
-        // This gives us a number of bits equal to the current count. The rest are all zeroed.
-        $bits = $bits << (self::PHP_INT_SIZE_BITS - $current);
-
-        // If the bits are all zero (equiv to all coin tosses = tails), then we increment the counter.
-        if ($bits == 0) {
-            return $current + 1;
+    public static function approximate_increment(int $current, int $increment = 1): int {
+        $approx = 2 ** $current;
+        // If we are incrementing by more than the current approximation, we can bump the count.
+        if ($increment > $approx) {
+            return self::approximate_increment($current + 1, $increment - $approx);
```

Comment:
Can this be done without recursion? Let’s say current = 4 so approx 16, and we increment by 128, then we are going g to turn either 7 most of the time and sometimes 8. So we can either do that maths directly, or call increment(log2($increment), $current ** 2) which should never recurse a second time